README.md

```diff
@@ -1 +1,491 @@
-# Ruby-Advanced-Building-Blocks-Enumerable-Methods
\ No newline at end of file
+# Ruby Advanced Building Blocks Enumerable Methods
+
+This is my solution to Microverse's tenth project. It is a custom implementation of some of the methods found on the Enumerable Module in Ruby.
+
+## Project Description
+
+The project description can be found [here](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#project-2-enumerable-methods).
+
+## Live Version
+
+The project can be tested [here](https://repl.it/@DavidAuza/Ruby-Advanced-Building-Blocks-Enumerable-Methods).
+
+## Project Author:
+
+- [David Auza](https://github.com/davidauza-engineer)
+
+## Code
+
+```ruby
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# This module contains an implementation of some of the methods found in the
+# Enumerable module
+module Enumerable
+
+  def my_each
+    # An enumerator is returned if no block is given
+    return to_enum unless block_given?
+
+    i = 0
+    self_class = self.class
+    array = self_class == Array ? self : flatten
+    while i < array.length
+      self_class == Array ? yield(array[i]) : yield(array[i], array[i + 1])
+      i += self_class == Array ? 1 : 2
+    end
+  end
+
+  def my_each_with_index
+    # If no block is given, an enumerator is returned instead.
+    return to_enum unless block_given?
+
+    array = self.class == Array ? self : to_a
+    i = 0
+    while i < length
+      yield(array[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    # If no block is given, an Enumerator is returned instead.
+    return to_enum unless block_given?
+
+    enumerable = self.class == Array ? [] : {}
+    if enumerable.class == Array
+      my_each do |n|
+        enumerable.push(n) if yield(n)
+      end
+    else
+      my_each do |key, value|
+        enumerable[key] = value if yield(key, value)
+      end
+    end
+    enumerable
+  end
+
+  def my_all?
+    # Return true if no block is given
+    return true unless block_given?
+
+    boolean = true
+    if self.class == Array
+      my_each do |n|
+        boolean = false unless yield(n)
+      end
+    else
+      my_each do |key, value|
+        boolean = false unless yield(key, value)
+      end
+    end
+    boolean
+  end
+
+  def my_any?
+    # Return true if no block is given
+    return true unless block_given?
+
+    boolean = false
+    if self.class == Array
+      my_each do |n|
+        boolean = true if yield(n)
+      end
+    else
+      my_each do |key, value|
+        boolean = true if yield(key, value)
+      end
+    end
+    boolean
+  end
+
+  def my_none?
+    # Return false in case no block is provided
+    return false unless block_given?
+
+    boolean = true
+    if self.class == Array
```

Comment:
```
%w{ant bear cat}.none? { |word| word.length == 5 } #=> true
%w{ant bear cat}.none? { |word| word.length >= 4 } #=> false
%w{ant bear cat}.none?(/d/)                        #=> true
[1, 3.14, 42].none?(Float)                         #=> false
[].none?                                           #=> true
[nil].none?                                        #=> true
[nil, false].none?                                 #=> true
[nil, false, true].none?                           #=> false
```

Reply:
Updated the method so it passes al those test cases 👍